### PR TITLE
New plotting of history and prediction

### PR DESCRIPTION
**Pull request by pleroy**

### CODE REVIEW

File: ksp_plugin_adapter/gl_lines.cs [hunk 1]
```diff
@@ -127,11 +127,118 @@ public enum Style {
...
     return PlanetariumCamera.Camera.WorldToScreenPoint(
                ScaledSpace.LocalToScaledSpace(world));
   }
 
+  private static XY ToScreen(XY rp2_point) {
+    UnityEngine.Camera camera = PlanetariumCamera.Camera;
+    return new XY{x = (rp2_point.x * camera.projectionMatrix[0, 0] + 1.0f) *
+                      0.5f * camera.pixelWidth,
+                  y = (rp2_point.y * camera.projectionMatrix[1, 1] + 1.0f) *
+                      0.5f * camera.pixelHeight};
```
Comment (eggrobin) [inline ksp_plugin_adapter/gl_lines.cs]: remove the fs on all those floating-point literals, they get promoted.

File: ksp_plugin_adapter/gl_lines.cs [hunk 2]
```diff
@@ -127,11 +127,118 @@ public enum Style {
...
+                                           UnityEngine.Color colour,
+                                           Style style) {
+    try {
+      UnityEngine.GL.Color(colour);
+
+      // First evaluate the total size of the lines.
+      int size = 0;
+      for (;
+            !rp2_lines_iterator.IteratorAtEnd();
+            rp2_lines_iterator.IteratorIncrement()) {
```
Comment (eggrobin) [inline ksp_plugin_adapter/gl_lines.cs]: Bad indent (2 lines).

File: ksp_plugin_adapter/gl_lines.cs [hunk 3]
```diff
@@ -127,11 +127,118 @@ public enum Style {
...
+                rp2_line_iterator.IteratorGetRP2LineXY());
+            if (previous_rp2_point.HasValue) {
+              if (style == Style.FADED) {
+                colour.a = (float)(4 * index + size) / (float)(5 * size);
+                UnityEngine.GL.Color(colour);
+              }
+              if (style != Style.DASHED || index % 2 == 1) {
+                UnityEngine.GL.Vertex3((float)previous_rp2_point.Value.x,
+                                        (float)previous_rp2_point.Value.y,
+                                        0);
```
Comment (eggrobin) [inline ksp_plugin_adapter/gl_lines.cs]: bad indent.

File: ksp_plugin_adapter/gl_lines.cs [hunk 4]
```diff
@@ -127,11 +127,118 @@ public enum Style {
...
+                UnityEngine.GL.Vertex3((float)current_rp2_point.x,
+                                        (float)current_rp2_point.y,
+                                        0);
```
Comment (eggrobin) [inline ksp_plugin_adapter/gl_lines.cs]: bad indent.

File: ksp_plugin_adapter/gl_lines.cs (hunk 2 above)
Comment (pleroy) [inline ksp_plugin_adapter/gl_lines.cs]: Done.

File: ksp_plugin_adapter/gl_lines.cs (hunk 3 above)
Comment (pleroy) [inline ksp_plugin_adapter/gl_lines.cs]: Done.

File: ksp_plugin_adapter/gl_lines.cs (hunk 4 above)
Comment (pleroy) [inline ksp_plugin_adapter/gl_lines.cs]: Done.

File: ksp_plugin_adapter/gl_lines.cs (hunk 1 above)
Comment (pleroy) [inline ksp_plugin_adapter/gl_lines.cs]: Done.

**Pull request merged**